src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,479 @@
+
+You can retrieve a `<Content />` component for use in your Astro files with `renderEntry`. For example, this page will render the contents of `content/announcements/welcome.md`:
+
+```astro "renderEntry"
+---
+// src/pages/welcome-announcement.astro
+import { renderEntry, getEntry } from 'astro:content';
+
+const announcementPost = await getEntry('announcements', 'welcome.md');
+// Pass a `getEntry` result or `getCollection` array item:
+const { Content } = await renderEntry(announcementPost);
+// Or pass a valid ID and collection name individually:
+const { Content } = await renderEntry({
+  id: announcementPost.id,
+  collection: announcementPost.collection,
+});
+---
+
+<h1>{announcementPost.data.title}</h1>
+<Content />
+```
+
+This example will render the contents of `content/announcements/welcome.md`.
+
+### Access content headings
+
+Astro [generates a list of headings](/en/guides/markdown-content/#exported-properties) for Markdown and MDX documents. You can access this list using the `headings` property from `renderEntry`:
+
+```astro "{ headings }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { headings } = await renderEntry(post);
+  const h1 = headings.find(h => h.depth === 1);
+  return <p>{h1}</p>
+})}
+```
+
+### Access injected frontmatter
+
+Astro allows you to [inject frontmatter using remark or rehype plugins.](/en/guides/markdown-content/#example-injecting-frontmatter) You can access these values using the `injectedFrontmatter` property from `renderEntry`:
+
+```astro "{ injectedFrontmatter }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { injectedFrontmatter } = await renderEntry(post);
+  return <p>{post.data.title} — {injectedFrontmatter.readingTime}</p>
+})}
+```
+
+Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `injectedFrontmatter` object.
+
+<details>
+<summary>**🙋 Why don't `getCollection` and `getEntry` contain these values?**</summary>
+
+The remark and rehype pipelines are only run when your content is **rendered.** This lets `renderEntry` access anything generated by these plugins like injected frontmatter. To stay performant, `getCollection` and `getEntry` do not have this capability.
+
+</details>
+
+## Generating pages from content collections
+
+You might map content collections to live URLs on your site. This should be  similar to globbing directories outside of `src/pages/`, [using `getStaticPaths` to generate routes dynamically.](/en/core-concepts/routing/#dynamic-routes)
```

Comment:
Not completely sold on keeping that last comparison bit but I think this is more goal focused: you can do this thing, here’s how, btw might be a bit like this other thing you once did.

I think the comparison might feel relevant today but in 6 months might already be dated, so could be worth saving for a announcement blog post for folks in the know.

```suggestion
You can create pages based on your content collections using [dynamic routes](/en/core-concepts/routing/#dynamic-routes). Use `getCollection()` inside your [`getStaticPaths()`](https://docs.astro.build/en/reference/api-reference/#getstaticpaths) function to build routes based on your collections. If you have used `Astro.glob()` to generate a dynamic route before, this might feel familiar.
```

Reply:
For an experimental release, I think most people will come with `Astro.glob` experience. The comparison is really helpful!